src/server/game/Handlers/MailHandler.cpp

```diff
@@ -585,11 +584,11 @@ void WorldSession::HandleGetMailList(WorldPackets::Mail::MailGetList& packet)
         if (m->state == MAIL_STATE_DELETED || cur_time < m->deliver_time)
             continue;
 
-        response.Mails.emplace_back(m, player);
-
         // max. 50 mails can be sent
-        if (response.Mails.size() >= 50)
-            break;
+        if (response.Mails.size() < 50)
```

Comment:
They are required changes.

Before TotalNumRecords included pending mails, this makes sure it does not. Making it include pending mails causes the client to complain about the mailbox being full.